test/src/test/java/hudson/security/CliAuthenticationTest.java

```diff
@@ -27,6 +37,19 @@
 
     @Rule
     public JenkinsRule j = new JenkinsRule();
+    
+    private SpySecurityListener spySecurityListener;
+
+    @Before
+    public void prepareListeners(){
+        //TODO simplify using #3021 into ExtensionList.lookupSingleton(SpySecurityListener.class)
+        this.spySecurityListener = ExtensionList.lookup(SecurityListener.class).get(SpySecurityListenerImpl.class);
+    }
+
+    @After
+    public void clearListener(){
+        spySecurityListener.clearPreviousCalls();
```

Comment:
Unnecessary. With `@Rule JenkinsRule`, each test case gets its own set of extensions created anew, and your list of events is correctly non`static`.